### PR TITLE
Make `qml.Qubitization` QJIT compatible

### PR DESCRIPTION
**Pull request by willjmax**
Catalyst does not support passing a list of wires into the identity operator, so the following code causes a segmentation fault `qml.Identity([0, 1])` To make this template QJIT compatible we need to build the Identity operator as a tensor product of single wire Identity operators instead of passing in a list of wires. This PR addresses [sc-74444].

### CODE REVIEW

File: pennylane/templates/subroutines/qubitization.py [hunk 1]
```diff
@@ -150,7 +150,11 @@ def compute_decomposition(*_, **kwargs):  # pylint: disable=arguments-differ
 
         decomp_ops = []
 
-        decomp_ops.append(qml.Reflection(qml.Identity(control)))
+        # QJIT is not compatible with qml.Identity(control), so we build the
+        # identity as a tensor product instead
+        identity = qml.prod(*[qml.Identity(wire) for wire in control])
```
Comment (obliviateandsurrender) [inline pennylane/templates/subroutines/qubitization.py:155]: Do we need to do this always or only when used in a `QJIT` setting?

Comment (willjmax) [reply pennylane/templates/subroutines/qubitization.py:155]: It's only needed for QJIT. Doing something like `qml.Identity(list_of_wires)` is not supported, so the identity operator needs to be built another way.

**Review by KetpuntoG — APPROVED** [review]
Looks good, thanks!

File: pennylane/templates/subroutines/qubitization.py [hunk 2]
```diff
@@ -150,7 +150,11 @@ def compute_decomposition(*_, **kwargs):  # pylint: disable=arguments-differ
 
         decomp_ops = []
 
-        decomp_ops.append(qml.Reflection(qml.Identity(control)))
+        # QJIT is not compatible with qml.Identity(control), so we build the
+        # identity as a tensor product instead
```
Comment (soranjh) [inline pennylane/templates/subroutines/qubitization.py:154]: No need to put this comment here. Instead, we should create an issue for Catalyst to support `qml.Identity(list_of_wires)`.

Comment (willjmax) [reply pennylane/templates/subroutines/qubitization.py:154]: I already submitted an issue [here](https://github.com/PennyLaneAI/catalyst/issues/1175).

**Review by obliviateandsurrender — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -68,6 +71,7 @@
   unprocessed diagonalizing gates.
   [(#6290)](https://github.com/PennyLaneAI/pennylane/pull/6290)
 
+
```
Comment (soranjh) [inline doc/releases/changelog-dev.md:74]: Please remove this blank line if not needed.

**Review by soranjh — APPROVED** [review]

**Pull request merged**